acme/acme/standalone.py

```diff
@@ -49,12 +56,83 @@ class ACMEServerMixin:  # pylint: disable=old-style-class
     allow_reuse_address = True
 
 
+class BaseDualNetworkedServers(object):
+    """Base class for a pair of IPv6 and IPv4 servers that tries to do everything
+       it's asked for both servers, but where failures in one server don't
+       affect the other.
+
+       If two servers are instantiated, they will serve on the same port.
+       """
+
+    def __init__(self, *args, **kwargs):
+        address = args[0]
+        port = address[1]
+        ServerClass = kwargs.pop("server_class", object)
```

Comment:
It can't be `socketserver.TCPServer`, because that doesn't like when you pass in an `ipv6` kwarg. I think it's actually fine to make it a named argument, because you probably wouldn't directly use the Base class anyway, and our `TLSSNI01DualNetworkedServers` and `HTTP01DualNetworkedServers` set that argument for you.